lib/instrumentation/undici.js

```diff
@@ -40,7 +70,7 @@ module.exports = function addUndiciChannels(agent, undici, modName, shim) {
    * @param {Object} params.request undici request object
    */
   diagnosticsChannel.channel('undici:request:create').subscribe(({ request }) => {
-    const parent = shim.getSegment()
+    const parent = getParentSegment()
```

Comment:
need to fix this line https://github.com/newrelic/node-newrelic/pull/937/files#diff-9fe41f6ad1e32c80844e1948f016e7d692ea907edd2620e8b9245c1c9da67b37R79. parent could be undefined